Standardize/Improve C# GD.Load() examples

wlsnmrk opened the pull request:
Following on from #7193, I've searched the docs for examples of `GD.Load()` that might not illustrate its requirement for absolute paths, fixed them to include a `res://` prefix, and made a few other changes:
* To standardize on the C# style, filenames and directories have been converted to PascalCase, except when loading GDScript files, and in one case a GLSL file (see below).
* I've changed to the typed version of `GD.Load()` (e.g., `GD.Load<PackedScene>()`) when examples were casting the result of an untyped call.
* I've used a `res://Path/To/FileName.extension` convention when the corresponding GDScript did not have an absolute path specified, to illustrate that the file might live in an arbitrary directory. When the GDScript example did have a `res://`, I've used the same absolute path.

Specific items I'd like to point up for review:
* `tutorials/shaders/compute_shaders.rst`: 
    * [I left a GLSL filename snake_cased](https://github.com/wlsnmrk/godot-docs/blob/45c6a1a4e4a5a1efddc6772c501e4f9411146f71/tutorials/shaders/compute_shaders.rst?plain=1#L187). I'm not a GLSL expert, but from some quick searching it seems like snake_case is kind of a de-facto standard. Happy to be corrected.
* `tutorials/best_practices/scenes_versus_scripts.rst`:
    * Required some significant changes in [the example for loading a script and assigning it to a `Node`](https://github.com/godotengine/godot-docs/compare/master...wlsnmrk:godot-docs:gd-load-fixes#diff-10a5f1b0589fa7ad1b08151c1dcd11bb5228644a9fbc421804f65bbece39b2d1R169-R183). I've verified that the updated version runs correctly in a test project, but wouldn't mind if someone double-checked my work.
        * Corresponding GDScript example was a `Node`, so I've changed C# from a `Resource` to a `Node`
        * `SetScript()` disposes the C# wrapper object, and the call comes before the node is in the tree, so I've added the object-ID workaround from https://github.com/godotengine/godot/issues/31994 to obtain a new reference
        * Moved the `AddChild()` call to before setting `Owner` on the child, per [documentation on `Owner`](https://docs.godotengine.org/en/stable/classes/class_node.html#class-node-property-owner)
    * [Changed a comment to hopefully make its meaning a little more clear](https://github.com/godotengine/godot-docs/compare/master...wlsnmrk:godot-docs:gd-load-fixes#diff-10a5f1b0589fa7ad1b08151c1dcd11bb5228644a9fbc421804f65bbece39b2d1R55), following the GDScript example.
* `tutorials/best_practices/godot_interfaces.rst`:
    * ~~I've left a field loading a snake_case GDScript rather than changing it to load a `.cs` script [in one example](https://github.com/wlsnmrk/godot-docs/blob/45c6a1a4e4a5a1efddc6772c501e4f9411146f71/tutorials/best_practices/godot_interfaces.rst?plain=1#L90), since it's loading from a (notional) add-on and those tend to be GDScript rather than C#. However, there doesn't seem to be a corresponding value in the GDSCript example, and I'm not sure if this example really needs to illustrate add-on directory layouts or cross-language scripting. Would be happy to change this to something else.~~ Changed below.
* `tutorials/scripting/resources.rst`:
    * [Changed a comment](https://github.com/godotengine/godot-docs/compare/master...wlsnmrk:godot-docs:gd-load-fixes#diff-9543c0c477dd21ce6356f57c792659ee49439568e90442bc16cf7df93be83461R90) to describe Godot loading the resource when the `GD.Load()` line is _executed_ rather than _read_. I'm not sure what the semantics of this are on the GDScript side, but I believe this to be more correct for C#; happy to be corrected.
    * [I've changed the filename "robi.png" to PascalCase](https://github.com/godotengine/godot-docs/compare/master...wlsnmrk:godot-docs:gd-load-fixes#diff-9543c0c477dd21ce6356f57c792659ee49439568e90442bc16cf7df93be83461R91) (i.e., "Robi.png"), in keeping with all the other changes. However, since it's referred to as "robi.png" in the body of the page, I'm not sure if we should make an exception to the naming convention here to avoid confusion.
* `tutorials/scripting/cross_language_scripting.rst`:
    * [Changed the path for loading a C# script from a GDScript to use PascalCase](https://github.com/godotengine/godot-docs/compare/master...wlsnmrk:godot-docs:gd-load-fixes#diff-7875f3ee292a53a639d2a0c22b902d99c0c356f9ad9339b1abee1f9209d0e2d8R77), to match the corresponding example for loading a GDScript from a C# script, which uses snake_case.

raulsntos commented on tutorials/best_practices/scenes_versus_scripts.rst:
```diff
@@ -39,8 +39,10 @@ a change in API:
     {
-        public static CSharpScript MyNode { get; } = GD.Load<CSharpScript>("MyNode.cs");
-        public static PackedScene MyScene { get; } = GD.Load<PackedScene>("MyScene.tscn");
+        public static CSharpScript MyNode { get; } = 
+            GD.Load<CSharpScript>("res://Path/To/MyNode.cs");
+        public static PackedScene MyScene { get; } = 
+            GD.Load<PackedScene>("res://Path/To/MyScene.tscn");
```
Trailing whitespace.

```suggestion
        public static CSharpScript MyNode { get; } =
            GD.Load<CSharpScript>("res://Path/To/MyNode.cs");
        public static PackedScene MyScene { get; } =
            GD.Load<PackedScene>("res://Path/To/MyScene.tscn");
```

raulsntos commented on tutorials/best_practices/scenes_versus_scripts.rst:
```diff
@@ -222,7 +228,8 @@ In the end, the best approach is to consider the following:
       public partial class Game : RefCounted
       {
-          public static PackedScene MyScene { get; } = GD.Load<PackedScene>("MyScene.tscn");
+          public static PackedScene MyScene { get; } = 
+              GD.Load<PackedScene>("res://Path/To/MyScene.tscn");
```
Trailing whitespace.

```suggestion
          public static PackedScene MyScene { get; } =
              GD.Load<PackedScene>("res://Path/To/MyScene.tscn");
```

raulsntos commented:
> I left a GLSL filename snake_cased. [...] it seems like snake_case is kind of a de-facto standard.

I don't know but I feel like a C# user would probably use PascalCase for those files too. But it's fine to leave it out of the PR for now.

> I've left a field loading a snake_case GDScript rather than changing it to load a `.cs` script in [one example](https://github.com/wlsnmrk/godot-docs/blob/45c6a1a4e4a5a1efddc6772c501e4f9411146f71/tutorials/best_practices/godot_interfaces.rst?plain=1#L90), since it's loading from a (notional) add-on and those tend to be GDScript rather than C#. However, there doesn't seem to be a corresponding value in the GDSCript example, and I'm not sure if this example really needs to illustrate add-on directory layouts or cross-language scripting.

I don't think this example is meant to show cross-language scripting, could probably be replaced with `res://Path/To/MyScript.cs` like the other members.

> I've changed the filename "robi.png" to PascalCase (i.e., "Robi.png"), in keeping with all the other changes. However, since it's referred to as "robi.png" in the body of the page, I'm not sure if we should make an exception to the naming convention here to avoid confusion.

I think there are plenty of C# examples in the documentation that do this already. For example, in [_Your first 2D game: Coding the player_](https://docs.godotengine.org/en/stable/getting_started/first_2d_game/03.coding_the_player.html) there are mentions to methods and signals by their GDScript name in the text, the screenshots also show a GDScript project.

It's true that this is confusing to users. We try to address it in the C# documentation, explaining that there are different naming conventions in C# and that they may have to do some "translation" when reading the class reference and tutorials. It's not ideal, but it's a complicated problem to solve (somewhat related issues about it: https://github.com/godotengine/godot-docs/issues/6622 https://github.com/godotengine/godot-docs/issues/6596).

wlsnmrk commented on tutorials/best_practices/scenes_versus_scripts.rst:
```diff
@@ -162,17 +166,19 @@ with it, and finally adds it as a child of the ``Main`` node:
-            Child.Owner = this;
+            var childID = Child.GetInstanceId();
+            Child.SetScript(GD.Load<Script>("res://Path/To/Child.cs"));
+            Child = (Node)GodotObject.InstanceFromId(childID);
```
Done; hopefully this is sufficient?

wlsnmrk commented on tutorials/best_practices/scenes_versus_scripts.rst:
```diff
@@ -162,17 +166,19 @@ with it, and finally adds it as a child of the ``Main`` node:
+            var childID = Child.GetInstanceId();
+            Child.SetScript(GD.Load<Script>("res://Path/To/Child.cs"));
+            Child = (Node)GodotObject.InstanceFromId(childID);
             AddChild(Child);
+            Child.Owner = this;
```
Good catch, thanks!

wlsnmrk commented:
> I don't know but I feel like a C# user would probably use PascalCase for those files too. But it's fine to leave it out of the PR for now.

Okay -- I've left it in snake_case for the time being.

> I don't think this example is meant to show cross-language scripting, could probably be replaced with `res://Path/To/MyScript.cs` like the other members.

Done, and changed the name of the field since `Library` didn't really make sense anymore.

> It's true that this is confusing to users. We try to address it in the C# documentation, explaining that there are different naming conventions in C# and that they may have to do some "translation" when reading the class reference and tutorials. It's not ideal, but it's a complicated problem to solve (somewhat related issues about it: #6622 #6596).

Fair enough; I'll leave it with the capital letter for the moment. It does seem like a difficult problem to solve generally, though I'm in agreement that a separate class reference at least would be very helpful.

raulsntos reviewed (approved):
Looks good to me, thanks!

AThousandShips commented on tutorials/best_practices/scenes_versus_scripts.rst:
```diff
@@ -50,8 +52,10 @@ a change in API:
             _myNode = MyNode.New().As<Node>();
-            _myScene = MyScene.Instantiate(); // Different. Instantiated from a PackedScene
-            _myInheritedScene = MyScene.Instantiate(PackedScene.GenEditState.Main); // Create scene inheriting from MyScene
+            // Different than calling new() or MyNode.New(). Instantiated from a PackedScene
```
```suggestion
            // Different than calling new() or MyNode.New(). Instantiated from a PackedScene.
```
Per the [comment style](https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html#comment-style-guide).

wlsnmrk replied:
Should I also fix the corresponding GDScript comments where applicable, or just stick to the updated C# comments?

AThousandShips replied:
Can do them too where applicable :)

wlsnmrk replied:
I started looking over the files I'd modified for comment formatting, and spotted a chance to get rid of some foo/bar examples in the cross-language scripting tutorial. However, those go hand-in-hand with several spots where a comment is used to indicate the expected output of a print line (the raw string, no quotes). ([Here's one example](https://github.com/godotengine/godot-docs/blob/9ce9593ee38b008ca325ec277118318e49c37bbc/tutorials/scripting/cross_language_scripting.rst?plain=1#L121).) 

Two questions:
* Should those instances be updated, and if so, what would be the appropriate format for that? Something like `Expected output is "My Value 1".`?
* If they should be updated, should I open another PR for that? Seems like this issue is drifting away from the original purpose here.

AThousandShips replied:
I'd say to deal with it separately indeed

Generally error prints are supposed to also end with a period, but it's far from uniform, and don't think it's relevant here, at least not to justify doing a sweeping fix

AThousandShips commented on tutorials/best_practices/scenes_versus_scripts.rst:
```diff
@@ -50,8 +52,10 @@ a change in API:
-            _myInheritedScene = MyScene.Instantiate(PackedScene.GenEditState.Main); // Create scene inheriting from MyScene
+            // Different than calling new() or MyNode.New(). Instantiated from a PackedScene
+            _myScene = MyScene.Instantiate();
+            // Create scene inheriting from MyScene
```
```suggestion
            // Create scene inheriting from MyScene.
```

AThousandShips commented on tutorials/best_practices/scenes_versus_scripts.rst:
```diff
@@ -155,24 +159,28 @@ with it, and finally adds it as a child of the ``Main`` node:
+            var childID = Child.GetInstanceId();
+            Child.SetScript(GD.Load<Script>("res://Path/To/Child.cs"));
+            // SetScript() causes the C# wrapper object to be disposed, so obtain a new
+            // wrapper for the Child node using its instance ID before proceeding
```
```suggestion
            // wrapper for the Child node using its instance ID before proceeding.
```

wlsnmrk commented:
Additional changes beyond the suggestions:
* Comment fixes in the same vein as the reviewed lines, in `singletons_autoload.rst` and `cross_language_scripting.rst`
* Realized the C# file name I introduced in `cross_langugage_scripting.rst` didn't match the type name, so fixed that
* Removed a GDScript line in `cross_language_scripting.rst` that referred to a nonexistent C# field.

I think there are more changes that could be made to bring the `cross_language_scripting.rst` C# and GDScript examples more in line with each other and their respective style guides, but I'll leave that for a different PR.

mhilbrunner commented:
Thanks, merged :)
